npm/webpack-dev-server-fresh/src/helpers/sourceRelativeWebpackModules.ts

```diff
@@ -0,0 +1,213 @@
+import Module from 'module'
+import path from 'path'
+import type { WebpackDevServerConfig } from '../devServer'
+
+type ModuleClass = typeof Module & {
+  _load(id: string, parent: Module, isMain: boolean): any
+  _resolveFilename(request: string, parent: Module, isMain: boolean, options?: { paths: string[] }): string
+  _cache: Record<string, Module>
+}
+
+export interface PackageJson {
+  name: string
+  version: string
+}
+
+export interface SourceRelativeWebpackResult {
+  framework?: {
+    importPath: string
+    packageJson: PackageJson
+  }
+  /**
+   * The webpack module instance
+   */
+  webpack: {
+    importPath: string
+    module: Function
+    packageJson: PackageJson
+    majorVersion: 4 | 5
+  }
+  /**
+   * The webpack dev-server instance
+   */
+  webpackDevServer: {
+    importPath: string
+    module: {
+      new (...args: unknown[]): unknown
+    }
+    packageJson: PackageJson
+    majorVersion: 3 | 4
+  }
+  /**
+   * html-webpack-plugin
+   */
+  htmlWebpackPlugin: {
+    importPath: string
+    module: unknown
+    packageJson: PackageJson
+    majorVersion: 4 | 5
+  }
+}
+
+const originalModuleLoad = (Module as ModuleClass)._load
+const originalModuleResolveFilename = (Module as ModuleClass)._resolveFilename
```

Comment:
Are we also going to bundle html-webpack-plugin v4 and v5, like we will do with `webpack-dev-server`? (are we still doing that? that's my understanding).

Reply:
Yes, this is done in the `package.json`:

```
"dependencies": {
  "html-webpack-plugin-4": "npm:html-webpack-plugin@^4",
  "html-webpack-plugin-5": "npm:html-webpack-plugin@^5",
```